docs/agent-checkpoints.md

```diff
@@ -0,0 +1,197 @@
+    "Checkpoint feature requires unique node names in the strategy metadata" 
+}
+```
+
+Make sure to set `uniqueNames = true` in your strategy metadata when using this feature.
+
+## Installation
+
+To use the AgentCheckpoint feature, you need to add it to your agent's configuration:
+
+```kotlin
+val agent = AIAgent {
+    // Agent configuration
+    
+    features {
+        install(AgentCheckpoint) {
+            // Configure the feature
+            snapshotProvider(InMemoryAgentCheckpointStorageProvider())
+            // Create checkpoints after each node (optional)
+            continuouslyPersistent() 
+        }
+    }
+}
+```
+
+## Configuration options
+
+The AgentCheckpoint feature has two main configuration options:
+
+### Storage provider
+
+Set the storage provider that will be used to save and retrieve checkpoints:
+
+```kotlin
+install(AgentCheckpoint) {
+    snapshotProvider(InMemoryAgentCheckpointStorageProvider())
+}
+```
+
+Available built-in providers:
+
+- `InMemoryAgentCheckpointStorageProvider`: in-memory storage (non-persistent)
+- `FileAgentCheckpointStorageProvider`: file-based storage
+- `NoAgentCheckpointStorageProvider`: no-op provider (default)
+
+### Continuous persistence
+
+Enable automatic checkpoint creation after each node execution:
+
+```kotlin
+install(AgentCheckpoint) {
+    continuouslyPersistent()
+}
+```
+
+When enabled, the agent will automatically create a checkpoint after each node is executed, allowing for fine-grained recovery.
+
+## Basic usage
+
+### Creating a checkpoint
+
+Create a checkpoint at a specific point in your agent's execution:
+
+```kotlin
+suspend fun example(context: AIAgentContextBase) {
+    // Create a checkpoint with the current state
+    val checkpoint = context.checkpoint().createCheckpoint(
+        agentId = context.id,
+        agentContext = context,
+        nodeId = "current-node-id",
+        lastInput = inputData
+    )
+    
+    // The checkpoint ID can be stored for later use
+    val checkpointId = checkpoint.checkpointId
+}
+```
+
+### Restoring from a checkpoint
+
+Restore an agent's state from a specific checkpoint:
+
+```kotlin
+suspend fun example(context: AIAgentContextBase, checkpointId: String) {
+    // Roll back to a specific checkpoint
+    context.checkpoint().rollbackToCheckpoint(checkpointId, context)
+    
+    // Or roll back to the latest checkpoint
+    context.checkpoint().rollbackToLatestCheckpoint(context)
+}
+```
+
+### Using extension functions
+
+The feature provides convenient extension functions for working with checkpoints:
+
+```kotlin
+suspend fun example(context: AIAgentContextBase) {
+    // Access the checkpoint feature
+    val checkpointFeature = context.checkpoint()
+    
+    // Or perform an action with the checkpoint feature
+    context.withCheckpoints(context) { ctx ->
+        // 'this' is the checkpoint feature
+        createCheckpoint(ctx.id, ctx, "node-id", inputData)
+    }
+}
+```
```

Comment:
```suggestion

```kotlin
suspend fun example(context: AIAgentContextBase) {
    // Access the checkpoint feature
    val checkpointFeature = context.persistency()
    
    // Or perform an action with the checkpoint feature
    context.withPersistency(context) { ctx ->
        // 'this' is the checkpoint feature
        createCheckpoint(ctx.id, ctx, "node-id", inputData)
    }
}
```
```